src/macros.rs

```diff
@@ -0,0 +1,238 @@
+macro_rules! setter {
+    ($buffer: ident, $name:ident, slice, $offset:expr) => {
+        impl<'a, T: AsRef<[u8]> + AsMut<[u8]> + ?Sized> $buffer<&'a mut T> {
+            $crate::pastey::item! {
+                pub fn [<$name _mut>](&mut self) -> &mut [u8] {
+                    &mut self.buffer.as_mut()[$offset]
+                }
+            }
+        }
+    };
+    ($buffer: ident, $name:ident, $ty:tt, $offset:expr) => {
+        impl<'a, T: AsRef<[u8]> + AsMut<[u8]>> $buffer<T> {
+            setter!($name, $ty, $offset);
+        }
+    };
+    ($name:ident, u8, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: u8) {
+                self.buffer.as_mut()[$offset] = value;
+            }
+        }
+    };
+    ($name:ident, u16, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: u16) {
+                use $crate::byteorder::{ByteOrder, NativeEndian};
+                NativeEndian::write_u16(&mut self.buffer.as_mut()[$offset], value)
+            }
+        }
+    };
+    ($name:ident, u32, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: u32) {
+                use $crate::byteorder::{ByteOrder, NativeEndian};
+                NativeEndian::write_u32(&mut self.buffer.as_mut()[$offset], value)
+            }
+        }
+    };
+    ($name:ident, u64, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: u64) {
+                use $crate::byteorder::{ByteOrder, NativeEndian};
+                NativeEndian::write_u64(&mut self.buffer.as_mut()[$offset], value)
+            }
+        }
+    };
+    ($name:ident, i8, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: i8) {
+                self.buffer.as_mut()[$offset] = value;
+            }
+        }
+    };
+    ($name:ident, i16, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: i16) {
+                use $crate::byteorder::{ByteOrder, NativeEndian};
+                NativeEndian::write_i16(&mut self.buffer.as_mut()[$offset], value)
+            }
+        }
+    };
+    ($name:ident, i32, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: i32) {
+                use $crate::byteorder::{ByteOrder, NativeEndian};
+                NativeEndian::write_i32(&mut self.buffer.as_mut()[$offset], value)
+            }
+        }
+    };
+    ($name:ident, i64, $offset:expr) => {
+        $crate::pastey::item! {
+            pub fn [<set_ $name>](&mut self, value: i64) {
+                use $crate::byteorder::{ByteOrder, NativeEndian};
+                NativeEndian::write_i64(&mut self.buffer.as_mut()[$offset], value)
+            }
+        }
+    };
+}
+
+#[macro_export(local_inner_macros)]
+macro_rules! buffer {
```

Comment:
Rust document said we should not use `local_inner_macros`.

[This is the only function we used in other crates. Is it possible to expose this macro only ](https://doc.rust-lang.org/reference/macros-by-example.html)

We actually only use this `buffer!` macro in other crates.

Reply:
So I should just use "cratre" right ?